### PR TITLE
Fixing knight ambiguous moves

### PR DESCRIPTION
**Pull request by jseverson**

### CODE REVIEW

**Comment by jseverson** [comment]
@marcosdsanchez i added a test but don't know how to run tests on my local for this sub repo. If you want to show me how i'll gladly run them and get the new one passing.

**Comment by jseverson** [comment]
@marcosdsanchez ok tests passing. please review when u get a chance. Then we can update our sha in chess.com repo

File: Tests/ChessGameTest.php [hunk 1]
```diff
@@ -244,6 +244,34 @@ public function testCompleteGame()
         $this->assertEquals($endFen, $this->game->renderFen());
     }
 
+    /**
+     * Test two knights that can both move to the same square, but one is pinned to the king.
+     *
+     * 1.e4 e5 2.Bc4 Nf6 3.d3 Nc6 4.Nc3 Bb4 5.Ne2
+     */
+    public function testAmbiguousKnightMoves()
```
Comment (marcosdsanchez) [inline Tests/ChessGameTest.php:252]: The problem I see with this test is that it passes in latest develop. The test should fail in latest develop and pass with your changes in this branch.

File: Tests/ChessGameTest.php [hunk 2]
```diff
@@ -244,6 +244,34 @@ public function testCompleteGame()
...
+    {
+        $endFen = 'r1bqk2r/pppp1ppp/2n2n2/4p3/1bB1P3/2NP4/PPP2PPP/R1BQK1NR w KQkq - 3 5';
```
Comment (marcosdsanchez) [inline Tests/ChessGameTest.php]: I think that the resulting fen should be this one: r1bqk2r/pppp1ppp/2n2n2/4p3/1bB1P3/2NP4/PPP1NPPP/R1BQK2R b KQkq - 4 5 (after Ne2 or Nge2 was played)

**Comment by jseverson** [comment]
@marcosdsanchez after i had to revert the commit i had to refix everything for some reason. it didnt rebase very well. anyhow...refixed in this PR. should be working and the test should fail pre fix and pass post fix.

**Comment by marcosdsanchez** [comment]
Thank you @jseverson

**Pull request merged**